### PR TITLE
Allow pipelines connected to "All messages" in content packs

### LINKED ISSUE #5439: Default stream must not added as entity to new content pack
When a pipeline is attached to the default stream the content pack will contain it as an entity, which then prevents installing it.

## Expected Behavior
When creating content pack with a pipeline and rule, it should be installable.

## Current Behavior
If the stream the pipeline is bound to already exists in the system, which is always the case for the default stream, then it cannot be installed at all.

## Steps to Reproduce (for bugs)
1. Create a pipeline, attach it to the default stream.
2. Create a content pack with that pipeline (it will add the default stream into the entity list).
3. Install the newly created content pack.
4. Note that it cannot be installed: an error is shown.

## Your Environment
* Graylog Version: 3.0.0-alpha.5

### PR DESCRIPTION
**Pull request by pschichtel**
## Description

This pull request prevents the content pack exporter from exporting clones of default streams when including pipelines that are connected to e.g. "All messages".

This closes #5439 and #5616.

## Motivation and Context

In the project I'm working we intend to bootstrap a graylog container from scratch without any manual configuration for consistent production deployments and automated updates. This requires us to include out "entrypoint pipeline" on "All messages" for initial routing. Currently this is not possible, as the import will fail.

## How Has This Been Tested?

As of now the changes have been tested in a local test environment.

## Types of changes

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

There is a test failure, at least on my system, that has been there before I changed anything and still failes. It would be great of someone else could verify. I updated the PipelineFacade test, but I haven't added specific tests for the changes.

### CODE REVIEW

**Comment by pschichtel** [comment]
Btw this also cleanly applied to 3.1

File: graylog2-server/src/main/java/org/graylog2/contentpacks/ContentPackService.java [hunk 1]
```diff
@@ -361,6 +361,10 @@ private ContentPackUninstallation uninstallContentPack(ContentPackInstallation i
 
         final ImmutableSet.Builder<Entity> entities = ImmutableSet.builder();
         for (EntityDescriptor entityDescriptor : resolvedEntities) {
+            // don't export default streams
```
Comment (kmerz) [inline graylog2-server/src/main/java/org/graylog2/contentpacks/ContentPackService.java]: Please remove the comment. Your code is self-explaining.

File: graylog2-server/src/main/java/org/graylog2/contentpacks/EntityDescriptorIds.java [hunk 2]
```diff
@@ -44,11 +46,22 @@ public static EntityDescriptorIds of(EntityDescriptor... entityDescriptors) {
 
     public static EntityDescriptorIds of(Collection<EntityDescriptor> entityDescriptors) {
         final ImmutableMap<EntityDescriptor, String> descriptorIds = entityDescriptors.stream()
-                .collect(ImmutableMap.toImmutableMap(Function.identity(), d -> UUID.randomUUID().toString()));
+                .collect(ImmutableMap.toImmutableMap(Function.identity(), d -> {
+                    // use the hardcoded IDs for default streams
```
Comment (kmerz) [inline graylog2-server/src/main/java/org/graylog2/contentpacks/EntityDescriptorIds.java]: Please remove unnecessary comment.

File: graylog2-server/src/main/java/org/graylog2/contentpacks/facades/PipelineFacade.java [hunk 3]
```diff
@@ -156,8 +160,20 @@ Entity exportNativeEntity(PipelineDao pipelineDao, EntityDescriptorIds entityDes
             final Object stream = nativeEntities.get(descriptor);
             if (stream instanceof Stream) {
                 streams.add((Stream) stream);
-            } else {
-                throw new MissingNativeEntityException(descriptor);
+            }
+            else {
```
Comment (kmerz) [inline graylog2-server/src/main/java/org/graylog2/contentpacks/facades/PipelineFacade.java]: please do not put else in a new line.

File: graylog2-server/src/main/java/org/graylog2/plugin/streams/Stream.java [hunk 4]
```diff
@@ -100,4 +100,8 @@ public static MatchingType valueOfOrDefault(String name) {
     String getIndexSetId();
 
     void setIndexSetId(String indexSetId);
+
+    static boolean isDefaultStream(String id) {
```
Comment (kmerz) [inline graylog2-server/src/main/java/org/graylog2/plugin/streams/Stream.java]: Since we give a `id` to check perhaps `isDefaultStreamId` is better?

**Comment by pschichtel** [comment]
@kmerz Yep, I agree with suggestions. Just pushed a commit to the branch.

**Comment by pschichtel** [comment]
The test is still missing though

**Comment by pschichtel** [comment]
I added tests based on the existing createNativeEntity and exportNativeEntity tests

**Pull request merged**